### PR TITLE
feat: Made get function available to all training jobs

### PR DESCRIPTION
**Pull request by ivanmkc**
Lifted `get` function to `_TrainingJob` so that all subclasses can use this function to get an existing TrainingJob.

Fixes #https://b.corp.google.com/issues/180746020

### CODE REVIEW

File: google/cloud/aiplatform/training_jobs.py [hunk 1]
```diff
@@ -103,6 +103,65 @@ def __init__(
...
+
+        Raises:
+            ValueError: If the retrieved training job's training task definition
+                doesn't match the custom training task definition.
+
+        Returns:
+            An AI Platform Training Job
+        """
+
+        # Create job with dummy parameters
```
Comment (sasha-gitg) [inline google/cloud/aiplatform/training_jobs.py:144]: I created a bug to extract project and location from resource name (b/180923329) if given using this parser: https://github.com/googleapis/python-aiplatform/blob/b7aeeb637b7c36792a2de0b65c868121a4a7e412/google/cloud/aiplatform_v1beta1/services/pipeline_service/client.py#L197

We can add as a follow up for all classes.

**Review by sasha-gitg — CHANGES_REQUESTED** [review]
Added a few comments. Thanks!

File: google/cloud/aiplatform/training_jobs.py [hunk 2]
```diff
@@ -103,6 +103,65 @@ def __init__(
         self._project = project
         self._gca_resource = None
 
+    @classmethod
+    @abc.abstractmethod
+    def _supported_training_schemas(self) -> Tuple[str]:
```
Comment (sasha-gitg) [inline google/cloud/aiplatform/training_jobs.py:108]:
```suggestion
    def _supported_training_schemas(cls) -> Tuple[str]:
```

Comment (sasha-gitg) [reply google/cloud/aiplatform/training_jobs.py:108]: The property decorator is appropriate here and is consistent with the abstract class property pattern we use in the SDK:

https://github.com/googleapis/python-aiplatform/blob/dev/google/cloud/aiplatform/base.py#L223

Concrete class usage:
https://github.com/googleapis/python-aiplatform/blob/dev/google/cloud/aiplatform/training_jobs.py#L72

**Review by sasha-gitg — CHANGES_REQUESTED** [review]

File: google/cloud/aiplatform/training_jobs.py (hunk 2 above)
Comment (ivanmkc) [reply google/cloud/aiplatform/training_jobs.py:108]: Ok I was wondering if they were compatible and took it off just to be safe as it didn't change too much.

I will add it back.

**Review by sasha-gitg — APPROVED** [review]
LGTM!

**Pull request merged**